src/app/visit/visit.component.ts

```diff
@@ -21,15 +28,15 @@ export class VisitComponent implements OnInit {
         'type',
         'image',
     ];
-    constructor(
-        private storeService: StoreService,
-        private snackService: SnackService,
-    ) {}
-
+    dataSource;
     ngOnInit(): void {
         this.patient = history.state;
         this.storeService.getStore().subscribe({
-            next: (data) => (this.store = data.sort((a, b) => a.id - b.id)),
+            next: (data) => (
+                (this.dataSource = data.sort((a, b) => a.id - b.id)),
+                (this.dataSource = new MatTableDataSource<Store>(data)),
```

Comment:
this
`this.dataSource = new MatTableDataSource<Store>(data)`
should be:
`this.dataSource = new MatTableDataSource<Store>(this.dataSource)`